### DISCUSSION TITLE
Customization

### DISCUSSION THREAD

**Discussion by borisdjartov** [rpreen/xcsf#130] [Ideas]
Hi everyone,

I really enjoy using the package. I was just wondering if there is a way to customize the LCS with python. I wanted to know if you can define your own selection mechanism and genetic operators? Additionally is there a way to define our own individual classifier type? For example, i would like a classifier that has a weight vector as the actions. Is there a way to do this? I think this is an excellent package and is incredibly fast however i think it has limits in terms of customizability.

Kind regards and keep up the great work!

**Comment by rpreen** [comment]
I don't think that it's possible to do any of those things from Python. The Python library is really just an interface to the underlying C library. 

To add/modify the selection mechanism it would involve modifying `ea.c`. 

The genetic operators are specific to each knowledge representation, so, for example, would involve modifying the mutation or crossover functions in `cond_rectangle.c` or perhaps adding a new function there to use information from other classifiers in the population.

To add new classifier types, it's pretty easy since the code is structured in an OO way (using virtual functions) so that would involve creating a new `.c` module that implements all of the functions for either a condition, action or prediction - existing ones can easily be used as a template and then it would just need to be added to the interface in either `condition.c` or `action.c` or `prediction.c`.
